scio-avro/src/main/scala/com/spotify/scio/avro/AvroIO.scala

```diff
@@ -267,20 +293,25 @@ final case class GenericRecordParseIO[T](path: String, parseFn: GenericRecord =>
 
 object AvroIO {
   object WriteParam {
-    private[avro] val DefaultNumShards = 0
-    private[avro] val DefaultSuffix = ""
-    private[avro] val DefaultCodec: CodecFactory = CodecFactory.deflateCodec(6)
-    private[avro] val DefaultMetadata: Map[String, AnyRef] = Map.empty
-    private[avro] val DefaultTempDirectory = null
+    private[scio] val DefaultNumShards = 0
+    private[scio] val DefaultSuffix = ""
+    private[scio] val DefaultCodec: CodecFactory = CodecFactory.deflateCodec(6)
+    private[scio] val DefaultMetadata: Map[String, AnyRef] = Map.empty
+    private[scio] val DefaultShardNameTemplate: String = null
```

Comment:
Another thought about consolidating some pipeline logic: what if we defined a single case class in scio-core, `WindowParams(shardNameTemplate: String, tempDirectory: String, filenamePolicyCreator: FilenamePolicyCreator)?`

Then either all of the `WriteParam` implementations could just include the single `WindowParams` instance; or in `ScioIO`, alongside `def write`, we could have an overloaded `def write(data: SCollection[T], params: WriteP, windowParams: WindowParams)` ... wdyt?

Reply:
None of these are strictly related to windowing; you can use them all in an unwindowed context. Maybe `FileWriteParams`? I guess `suffix` would also end up in this class. 

It'd help somewhat for consistency across IOs but we'd still (at least per our current `saveAs*` design) need to replicate the parameters to all the save methods. It seems like only the very earliest `saveAs` methods took the entire `WriteParam` class as an argument, and IMO those should be migrated to the current status quo if there's any left.